### PR TITLE
Fix obfuscate failed by extracting SwiftFileList file

### LINKED ISSUE #64: Xcode 11 beta5 obfuscate failed
Hi, 
I found this tool not work on Xcode 11.0 beta 5. The job will be failed with follow error:
```sh
SwiftShield 3.3.3
Verbose Mode
Automatic mode
Building project to gather modules and compiler arguments...
Found module Unbox
Found module Cartography
Found module SwiftShieldExample
-- Indexing Unbox-OutputFileMap.json --
ERROR: Could not index Unbox-OutputFileMap.json, aborting. SK Error: '/Users/*****/Library/Developer/Xcode/DerivedData/SwiftShieldExample-flbvboqmqhugnfetbofsyikoufak/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/Unbox.build/Objects-normal/armv7/Unbox-OutputFileMap.json' is not part of the input files
```

Through some code check and debug, I found 
```swift

/// func getModulesAndCompilerArguments() -> [Module] 
let path = "/usr/bin/xcodebuild"
let projectParameter = isWorkspace ? "-workspace" : "-project"
let arguments: [String] = [projectParameter, projectToBuild, "-scheme", schemeToBuild]
let cleanTask = Process()
cleanTask.launchPath = path
cleanTask.arguments = ["clean", "build"] + arguments
let outpipe: Pipe = Pipe()
cleanTask.standardOutput = outpipe
cleanTask.standardError = outpipe
cleanTask.launch()
let outdata = outpipe.fileHandleForReading.readDataToEndOfFile()
```
The `outdata` is different with the Xcode 10's behavior, and the result of `parseCompilerArguments` function is different too.
```
/// Xcode 11
- 0 : "/usr/bin/swiftc"
- 1 : "-incremental"
- 2 : "-module-name"
- 3 : "ObfuscatorDemo"
- 4 : "-Onone"
- 5 : "-enable-batch-mode"
- 6 : "-enforce-exclusivity=checked"
- 7 : "@/Users/xxxx/Library/Developer/Xcode/DerivedData/ObfuscatorDemo-ctlepdgprmrxjydyctthxzvqkvid/Build/Intermediates.noindex/ObfuscatorDemo.build/Debug-iphoneos/ObfuscatorDemo.build/Objects-normal/arm64/ObfuscatorDemo.SwiftFileList"
- 8 : "-DDEBUG"
- 9 : "-sdk"
- 10 : "/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.0.sdk"
- 11 : "-target"
- 12 : "arm64-apple-ios12.2"
- 13 : "-g"
- 14 : "-module-cache-path"
- 15 : "/Users/xxxx/Library/Developer/Xcode/DerivedData/ModuleCache.noindex"

/// Xcode 10.2
- 3 : "ObfuscatorDemo"
- 4 : "-Onone"
- 5 : "-enable-batch-mode"
- 6 : "-enforce-exclusivity=checked"
- 7 : "-DDEBUG"
- 8 : "-sdk"
- 9 : "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk"
- 10 : "-target"
- 11 : "arm64-apple-ios12.2"
- 12 : "-g"
- 13 : "-module-cache-path"
- 14 : "/Users/xxxx/Library/Developer/Xcode/DerivedData/ModuleCache.noindex"
- 15 : "-Xfrontend"
- 16 : "-serialize-debugging-options"
- 17 : "-embed-bitcode-marker"
- 18 : "-enable-testing"
- 19 : "-index-store-path"
- 20 : "/Users/xxxx/Library/Developer/Xcode/DerivedData/ObfuscatorDemo-ctlepdgprmrxjydyctthxzvqkvid/Index/DataStore"
- 21 : "-swift-version"
- 22 : "5"
- 23 : "-I"
- 24 : "/Users/xxxx/Library/Developer/Xcode/DerivedData/ObfuscatorDemo-ctlepdgprmrxjydyctthxzvqkvid/Build/Products/Debug-iphoneos"
- 25 : "-F"
- 26 : "/Users/xxxx/Library/Developer/Xcode/DerivedData/ObfuscatorDemo-ctlepdgprmrxjydyctthxzvqkvid/Build/Products/Debug-iphoneos"
- 27 : "-c"
- 28 : "-j12"
- 29 : "/Users/xxxx/git-repo/ObfuscatorDemo/ObfuscatorDemo/ViewController.swift"
- 30 : "/Users/xxxx/git-repo/ObfuscatorDemo/ObfuscatorDemo/TestViewController.swift"
- 31 : "/Users/xxxx/git-repo/ObfuscatorDemo/ObfuscatorDemo/XibView.swift"
- 32 : "/Users/xxxx/git-repo/ObfuscatorDemo/ObfuscatorDemo/AppDelegate.swift"
```

On line 7, a file with `SwiftFileList` extension appears, and the swift source files are dismissed on Line 29-32. `xxx.SwiftFileList` file contains all swift source files' path in the project.

With my demo project,  If I get the items from this file and append to `CompilerArguments`, delete the `.SwiftFile` item from origin `CompilerArguments`, this job would be successful.

I'm not sure if this solution is correct. And is this the feature of SourceKit or Xcode 11.

### PR DESCRIPTION
**Pull request by ldp940622**
This PR wants to fix #64 and make tool compatible with Xcode11.
However, I noticed that the Circle.CI seems not worked for this repo. 
Maybe we need a better way to test these changes on CI.

### CODE REVIEW

File: swiftshield-Sources/XcodeProjectBuilder.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ final class XcodeProjectBuilder {
         let arguments: [String] = [projectParameter, projectToBuild, "-scheme", schemeToBuild]
         let cleanTask = Process()
         cleanTask.launchPath = path
-        cleanTask.arguments = ["clean", "build"] + arguments
+        cleanTask.arguments = ["clean", "build"] + arguments + ["-sdk", "iphonesimulator"]
```
Comment (rockbruno) [inline swiftshield-Sources/XcodeProjectBuilder.swift]: Is this necessary on Xcode 11? What happens if you have a watchOS target on your app?

Comment (ldp940622) [reply swiftshield-Sources/XcodeProjectBuilder.swift]: Without `-sdk` parameter, the build will be failed with error
```
Code Signing Error: "SwiftShieldExample" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor.
Code Signing Error: Code signing is required for product type 'Application' in SDK 'iOS 13.0'
```
Of course most of projects should contain these items. But if we want to obfuscate the demo project and without code sign and profiles, this job will not be successful.

Comment (ldp940622) [reply swiftshield-Sources/XcodeProjectBuilder.swift]: OK, I have reverted these changes.

**Comment by rockbruno** [comment]
Thanks a lot for working on this! I'll update the READMEs separately.

**Pull request merged**